avocado/core/messages.py

```diff
@@ -359,16 +360,18 @@ class StdoutMessageHandler(BaseRunningMessageHandler):
 
     def handle(self, message, task, job):
         self._save_to_default_file(message, task)
-        self._save_message_to_file(
-            "stdout", message["log"], task, message.get("encoding", None)
-        )
+        if not message.get("log_only", False):
+            self._save_message_to_file(
+                "stdout", message["log"], task, message.get("encoding", None)
+            )
 
 
 class StderrMessageHandler(BaseRunningMessageHandler):
     """
     Handler for stderr message.
 
-    It will save the stderr to the stderr and debug file in the task directory.
+    It will save the stderr to the debug file in the task directory,
+    and optionally (depending on "log_only" value) to the stderr file.
```

Comment:
Same here.

Reply:
Done for both in 61aae13a28a40f8d73f87cd833e7fc4e36342410 (force pushed).